MyCryptoHQ/MyCrypto
# Check Transaction page (Pt. 1 - The Basics)

Addresses the first part of #775, the check transaction page and component.

## Description

This implements the ability to search for a transaction by hash, and get some information about it. To keep changes small, this is only the transaction status part. No cancel functionality is a part of this (yet.) In addition, a few improvements have been made to the amount of information given by also fetching `eth_getTransactionReceipt` on mined transactions. Anything Etherscan shows that this doesn't show is because the information isn't available purely from JSON RPC api calls.

## Changes

* Adds a "Check TX" navigation item and page
* Adds `transactions` actions / reducer / saga for dealing with existing transactions
    * I was a little worried about this getting confused with the `transaction` ones, but couldn't come up with a different name. Maybe that should be renamed to `send`?
* Adds a stand-alone component called `TransactionStatus` that takes in a tx hash, kicks off a request for data, and renders that when it comes in.
    * Displays the following fields:
    * Status, which is SUCCESFUL, PENDING, FAILED, or UNKNOWN (For pre-byzantium transactions that don't have the `status` field.)
    * TX Hash (Linked to block explorer)
    * Block # when mined (Linked to block explorer)
    * From address (Linked to block explorer)
    * To address (Linked to block explorer)
    * Amount
    * Gas price
    * Gas limit
    * Gas used when mined
    * Transaction fee in network unit (gas used * gas price)
    * Deployed contract address (if applicable)
    * Nonce
    * Input Data
* Some of the copy from the old version was left out due to not having features / it being kinda verbose

## Future Features

1. Auto-reload for pending transactions
2. Identify contract sends (At least ERC-20) and show token values instead
3. Transaction canceling like prod
4. Re-send transaction where it auto-fills the transaction form with the past data
5. Maybe rendering the `TransactionStatus` component on Send, after you send, with auto-reloading.

## Steps to Test

1. Start the app on the `check-tx` branch
2. Go to `/tx-status`
3. Enter transaction hashes from https://etherscan.io/txs and confirm their information matches
4. Enter transaction hashes from https://etherscan.io/txsPending and confirm their information matches (and they show up as pending.)
5. Try to enter an invalid transaction and be unable to submit

## Screenshots

### The Form

![screen shot 2018-02-15 at 2 08 30 pm](https://user-images.githubusercontent.com/649992/36277209-435e8db2-125e-11e8-9de4-733faad8f054.png)

### If You Enter an Address Instead

![screen shot 2018-02-15 at 2 08 37 pm](https://user-images.githubusercontent.com/649992/36277218-4a547154-125e-11e8-9c2c-6117c00bc9d9.png)

### Succesful TX

![screen shot 2018-02-15 at 2 06 55 pm](https://user-images.githubusercontent.com/649992/36277252-5dedbc8e-125e-11e8-87c2-47fc03a2c024.png)


### Failed TX

![screen shot 2018-02-15 at 2 07 21 pm](https://user-images.githubusercontent.com/649992/36277243-5874bdc0-125e-11e8-8631-bb5c4f84b164.png)


### Pending TX

![screen shot 2018-02-15 at 2 07 45 pm](https://user-images.githubusercontent.com/649992/36277233-50f8c046-125e-11e8-8608-a9904c2bdc52.png)


### Creating a New Contract

![screen shot 2018-02-15 at 2 37 28 pm](https://user-images.githubusercontent.com/649992/36277193-30b4688a-125e-11e8-90f0-5ee726bc3677.png)
